ArduCopter/mode_auto.cpp

```diff
@@ -802,7 +802,10 @@ void ModeAuto::takeoff_run()
     if ((copter.g2.auto_options & (int32_t)Options::AllowTakeOffWithoutRaisingThrottle) != 0) {
         copter.set_auto_armed(true);
     }
-    auto_takeoff_run();
+    if (auto_takeoff_run()) {
+        wp_nav->shift_wp_origin_and_destination_to_stopping_point_xy();
```

Comment:
why call wp_nav here?

Reply:
This only runs when we reach our target height.